lib/debezium/schema_test.go

```diff
@@ -82,203 +79,85 @@ func TestField_GetScaleAndPrecision(t *testing.T) {
-		},
+	{
+		// Floats
+		assert.Equal(t, typing.Float, Field{Type: Float}.ToKindDetails())
+		assert.Equal(t, typing.Float, Field{Type: Double}.ToKindDetails())
+	}
+	{
+		// Decimals
 		{
-			name: "Debezium Map",
-			field: Field{
-				DebeziumType: "",
-				Type:         "map",
-			},
-			expectedKindDetails: typing.Struct,
-		},
+			assert.Equal(
+				t, typing.NewDecimalDetailsFromTemplate(typing.EDecimal, decimal.NewDetails(10, 5)),
+				Field{DebeziumType: KafkaDecimalType, Parameters: map[string]any{"scale": 5, KafkaDecimalPrecisionKey: 10}}.ToKindDetails(),
+			)
+		}
 		{
-			name:                "UUID",
-			field:               Field{DebeziumType: UUID, Type: String},
-			expectedKindDetails: typing.String,
-		},
+			// Variable numeric decimal
+			assert.Equal(
+				t, typing.NewDecimalDetailsFromTemplate(typing.EDecimal, decimal.NewDetails(decimal.PrecisionNotSpecified, decimal.DefaultScale)),
+				Field{DebeziumType: KafkaVariableNumericType, Parameters: map[string]any{"scale": 5}}.ToKindDetails(),
+			)
+		}
+	}
+	{
+		// String
+		assert.Equal(t, typing.String, Field{Type: String}.ToKindDetails())
+	}
+	{
+		// Bytes
+		assert.Equal(t, typing.String, Field{Type: Bytes}.ToKindDetails())
 	}
+	{
+		// UUID
+		assert.Equal(t, typing.String, Field{DebeziumType: UUID, Type: String}.ToKindDetails())
+	}
+	{
+		// Structs
+		assert.Equal(t, typing.Struct, Field{Type: Struct}.ToKindDetails())
+		assert.Equal(t, typing.Struct, Field{Type: Map}.ToKindDetails())
 
-	for _, tc := range tcs {
-		assert.Equal(t, tc.expectedKindDetails, tc.field.ToKindDetails(), tc.name)
+		assert.Equal(t, typing.Struct, Field{DebeziumType: JSON}.ToKindDetails())
+	}
+	{
+		// Booleans
+		assert.Equal(t, typing.Boolean, Field{Type: Boolean}.ToKindDetails())
+	}
+	{
+		// Array
+		assert.Equal(t, typing.Array, Field{Type: Array}.ToKindDetails())
+	}
+	{
+		// Invalid
+		assert.Equal(t, typing.Invalid, Field{Type: "unknown"}.ToKindDetails())
+		assert.Equal(t, typing.Invalid, Field{Type: ""}.ToKindDetails())
+	}
+	{
+		// Timestamp
+		for _, dbzType := range []SupportedDebeziumType{Timestamp, DateTimeKafkaConnect, MicroTimestamp, NanoTimestamp} {
+			assert.Equal(t, typing.NewKindDetailsFromTemplate(typing.ETime, ext.DateTimeKindType), Field{DebeziumType: dbzType}.ToKindDetails())
+		}
+	}
+	{
+		// Dates
+		for _, dbzType := range []SupportedDebeziumType{Date, DateKafkaConnect} {
+			assert.Equal(t, typing.NewKindDetailsFromTemplate(typing.ETime, ext.DateKindType), Field{DebeziumType: dbzType}.ToKindDetails())
+		}
+	}
+	{
+		// Time
+		for _, dbzType := range []SupportedDebeziumType{Time, TimeKafkaConnect, MicroTime, NanoTime, TimeWithTimezone} {
+			assert.Equal(t, typing.NewKindDetailsFromTemplate(typing.ETime, ext.TimeKindType), Field{DebeziumType: dbzType}.ToKindDetails())
+		}
+	}
+	{
+		// Datetime
+		assert.Equal(t, typing.NewKindDetailsFromTemplate(typing.ETime, ext.DateTimeKindType), Field{DebeziumType: DateTimeWithTimezone}.ToKindDetails())
```

Comment:
Could this one go in the array on line 143?

Reply:
Yeah we can put it in there. When we delineate support for TIMESTAMP and DATETIME, we can then pull it out